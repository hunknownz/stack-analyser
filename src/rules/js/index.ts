import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
