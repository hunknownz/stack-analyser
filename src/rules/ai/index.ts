import './anthropic.js';
import './aws.bedrock.js';
import './aws.comprehend.js';
import './aws.polly.js';
import './aws.rekognition.js';
import './aws.sagemaker.js';
import './aws.transcribe.js';
import './aws.translate.js';
import './azure.openai.js';
import './cohere.js';
import './deepseek.js';
import './gcp.aiplatform.js';
import './gcp.dialogflow.js';
import './gcp.language.js';
import './gcp.speech.js';
import './gcp.translate.js';
import './gcp.vertex.js';
import './gcp.vision.js';
import './geminiai.js';
import './groq.js';
import './huggingface.js';
import './mistralai.js';
import './ollama.js';
import './openai.js';
import './perplexityai.js';
import './vercel.ai.js';
import './xai.js';
