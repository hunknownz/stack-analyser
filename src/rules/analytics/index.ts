import './amplitude.js';
import './googleanalytics.js';
import './vercel.analytics.js';
