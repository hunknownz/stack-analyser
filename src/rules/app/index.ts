import './caddy.js';
import './discourse.js';
import './grafana.js';
import './httpd.js';
import './kibana.js';
import './kong.js';
import './nginx.js';
import './traefik.js';
import './zookeeper.js';
