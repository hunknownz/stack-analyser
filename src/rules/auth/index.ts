import './auth0.js';
import './aws.cognito.js';
import './betterauth.js';
import './clerk.js';
import './frontegg.js';
import './fusionauth.js';
import './logto.js';
import './okta.js';
import './orysh.js';
import './stytch.js';
import './supabase.auth.js';
import './workos.js';
