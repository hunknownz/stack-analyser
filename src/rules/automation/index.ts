import './browerbase.js';
import './browseruse.js';
import './firecrawl.js';
import './inngest.js';
import './n8n.js';
