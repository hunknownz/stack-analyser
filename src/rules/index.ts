import './ai/index.js';
import './analytics/index.js';
import './auth/index.js';
import './app/index.js';
import './automation/index.js';
import './ci/index.js';
import './cms/index.js';
import './collaboration/index.js';
import './communication/index.js';
import './cloud/index.js';
import './crm/index.js';
import './db/index.js';
import './etl/index.js';
import './framework/index.js';
import './hosting/index.js';
import './iac/index.js';
import './iconset/index.js';
import './language/index.js';
import './linter/index.js';
import './maps/index.js';
import './monitoring/index.js';
import './notification/index.js';
import './network/index.js';
import './payment/index.js';
import './package_manager/index.js';
import './orm/index.js';
import './queue/index.js';
import './runtime/index.js';
import './saas/index.js';
import './security/index.js';
import './spec/index.js';
import './ssg/index.js';
import './storage/index.js';
import './test/index.js';
import './tool/index.js';
import './ui/index.js';
import './ui_framework/index.js';
