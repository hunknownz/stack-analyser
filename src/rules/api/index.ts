import './aws.kinesis.js';
import './aws.rekognition.js';
import './aws.sagemaker.js';
import './gcp.aiplatform.js';
import './gcp.dialogflow.js';
import './gcp.kms.js';
import './gcp.language.js';
import './gcp.maps.js';
import './gcp.speech.js';
import './gcp.translate.js';
import './gcp.vision.js';
import './sitecore.xmcloud.js';
import './vercel.ai.js';
