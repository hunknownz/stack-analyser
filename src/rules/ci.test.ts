import { describe, expect, it } from 'vitest';

import { analyser } from '../analyser/index.js';
import { listIndexed } from '../common/techs.js';
import { FakeProvider } from '../provider/fake.js';
import { rawList } from '../rules.js';
import './index.js';

const paths: string[] = [];
for (const item of rawList) {
  if (
    item.type !== 'file' ||
    listIndexed[item.ref.tech].type !== 'ci' ||
    !item.ref.files
  ) {
    continue;
  }

  paths.push('example' in item.ref ? item.ref.example : item.ref.files[0]);
}

describe('ci', () => {
  it('should match everything', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': paths,
        },
        files: {},
      }),
    });
    expect(res.toJson('').techs).toStrictEqual([
      'appveyor',
      'browserstack',
      'circleci',
      'codesandboxci',
      'cypressci',
      'dependabot',
      'githubactions',
      'gitlabci',
      'javascript',
      'jenkins',
      'relativeci',
      'renovate',
      'teamcity',
      'travisci',
    ]);
  });

  it('enforce that we match .github', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['.github/'],
          '/.github': ['workflows/'],
          '/.github/workflows': ['main.yml'],
        },
        files: {
          '/.github/workflows/main.yml': '',
        },
      }),
    });
    expect(res.toJson('').techs).toStrictEqual(['github', 'githubactions']);
  });
});
