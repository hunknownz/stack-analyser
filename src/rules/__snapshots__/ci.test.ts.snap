// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ci > should match everything 1`] = `
[
  "appveyor",
  "azure.ci",
  "browserstack",
  "circleci",
  "cirrusci",
  "cloudbees.codeship",
  "codeclimate",
  "codecov",
  "codesandboxci",
  "cypressci",
  "deepsource",
  "dependabot",
  "droneci",
  "github.actions",
  "gitlab.ci",
  "javascript",
  "jenkins",
  "relativeci",
  "renovate",
  "sonarqube",
  "styleci",
  "teamcity",
  "travisci",
]
`;
